src/unstract/sdk/file_storage/fs_permanent.py

```diff
@@ -75,7 +83,9 @@ def read(
         """
         try:
             # Lazy copy to the destination/remote file system
-            self._copy_on_write(path)
+            self._copy_on_read(path, legacy_storage_path)
```

Comment:
@gaya3-zipstack Instead of making the check for `legacy_storage_path` for None in `_copy_on_write`, it can be made before invoking the method, and remove the checks in method. 
Would avoid extra function calls.

Ref :
```
if legacy_storage_path:
            self._copy_on_read(path, legacy_storage_path)
```

Reply:
Done.